include/deal.II/numerics/data_out_dof_data.templates.h

```diff
@@ -1212,21 +1212,27 @@ DataOut_DoFData<DoFHandlerType, patch_dim, patch_space_dim>::
   if (actual_type == type_dof_data)
     {
       // output vectors for which at least part of the data is to be interpreted
-      // as vector fields cannot be complex-valued, because we cannot visualize
-      // complex-valued vector fields
-      Assert(!((std::find(
-                  data_component_interpretation.begin(),
-                  data_component_interpretation.end(),
-                  DataComponentInterpretation::component_is_part_of_vector) !=
-                data_component_interpretation.end()) &&
-               new_entry->is_complex_valued()),
-             ExcMessage(
-               "Complex-valued vectors added to a DataOut-like object "
-               "cannot contain components that shall be interpreted as "
-               "vector fields because one can not visualize complex-valued "
-               "vector fields. However, you may want to try to output "
-               "this vector as a collection of scalar fields that can then "
-               "be visualized by their real and imaginary parts separately."));
+      // as vector or tensor fields cannot be complex-valued, because we cannot
+      // visualize complex-valued vector fields
+      Assert(
+        !(((std::find(
+              data_component_interpretation.begin(),
+              data_component_interpretation.end(),
+              DataComponentInterpretation::component_is_part_of_vector) !=
+            data_component_interpretation.end()) ||
+           (std::find(
```

Comment:
In particular, your condition is `! (A || B)`, which is the same as `!A && !B`, and because you're now enforcing two conditions, they can be checked in separate assertions.